packages/js/product-editor/src/blocks/input/block.json

```diff
@@ -0,0 +1,41 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "woocommerce/product-input-field",
+	"title": "Product input control",
+	"category": "woocommerce",
+	"description": "A input control for use in the product editor.",
+	"keywords": [ "products", "input" ],
+	"textdomain": "default",
+	"attributes": {
+		"label": {
+			"type": "string",
+			"__experimentalRole": "content"
+		},
+		"property": {
+			"type": "string"
+		},
+		"placeholder": {
+			"type": "string"
+		},
+		"required": {
+			"type": "boolean"
+		},
+		"validationRegex": {
+			"type": "string"
+		},
```

Comment:
It would be nice to have some higher-level support for common types of text. These could be handled via an attribute on this block type (`type`?), or by even more specific block types (`woocommerce/product-email-field`, `woocommerce/product-phone-number-field`, etc.).

Bottom line, ideally devs shouldn't have to input regexes for common things...

- email
- phone (this one would be tricky, since formats vary between locales, but all the more reason to provide support out of the box rather than have any developer have to re-implement it)

And, some very common types of validation:

- min/max length
- unique

Reply:
I propose we keep this block as generic as possible and work on implementing separate blocks for specific needs. As an example, the email validation itself is very tricky, as well as the phone validation, as you mentioned. 

I will work on adding the mentioned common types of validation.